doc/acl/egress_outer_dscp_change_table.md

```diff
@@ -0,0 +1,505 @@
+- Specific ACL table port bindings: Since the packet can come from a few set of ports but can go out on any of the ports available, the MARK_META/V6 tables need to be bound to only the ports from which the packet is expected. The EGR_SET_DSCP would be bound to all the ports except loopback and internal ports.
+
+##### Cons
+This mechanism has a major disadvantage from the user's perspective. They would need to keep track of the Metadata availability and range. They would also need to keep track of the relationship between MARK_META/V6 entries and EGR_SET_DSCP entries as one entry of EGR_SET_DSCP may be referenced by multiple entries from the other tables. This would expose the user to the internal functioning of the ACL metadata capability.
+
+
+#### Option-B: Augment existing L3/L3V6 tables to include support for metadata action type.
+
+This approach is similar to Option A with the exception that instead of creating new Ingress tables such as MARK_META/V6, the existing L3 and L3V6 tables would be augmented to include SET_METADATA action field. This would still require the creation of an EGR_SET_DSCP table. This has the advantages of Option A with less ingress stage TCAM utilization.
+
+##### User accessible tables
+    /*
+     * L3 table augmentation
+     * |----------------------------------------------------|
+     * |                        |              |            |
+     * |    Match Type          |   Actions    |    Bind    |
+     * |                        |              |   points   |
+     * |----------------------------------------------------|
+     * | No additionnal Match | SET_METADATA*|              |
+     * |----------------------------------------------------|
+     
+     * L3V6 Table augmentation
+     * |----------------------------------------------------|
+     * |                        |              |            |
+     * |    Match Type          |   Actions    |    Bind    |
+     * |                        |              |   points   |
+     * |----------------------------------------------------|
+     * | No additionnal Match | SET_METADATA*|              |
+     * |----------------------------------------------------|
+
+     * EGR_SET_DSCP
+     * |----------------------------------------------------|
+     * |                        |              |            |
+     * |    Match Type          |   Actions    |    Bind    |
+     * |                        |              |   points   |
+     * |----------------------------------------------------|
+     * |  MATCH_METADATA        | SET_DSCP     |    PORT    |
+     * |                        |              |    LAG     |
+     * |----------------------------------------------------|
+     */
+
+
+##### Pros
+- Ease of Implementation: The changes to the Orchagent would be minimal.
+- The TCAM utilization would be minimal with the introduction of just one egress table. This table would be associated with all the ports excluding loopback and internal ports.
```

Comment:
The idea that the TCAM can be shared between L3/L3V6 rules and the MARK\_META/METAV6 rules is not always true. This is because either L3/L3V6 rules can shadow MARK\_META/METAV6 rules or the other way.

Reply:
no, this is not the intent here. The Egress SET DSCP table only sets the DSCP based on the metadata value. the metadata value would be set by the ingress L4 or L6 tables. Here single egress table means that we dont need to have 2 separate egress set DSCP table for V4 and V6.